### ISSUE TITLE
Do not set SIGTERM handler in Task.__call__()

### ISSUE THREAD

**Issue by rneilson** [celery/celery#5775]
PR #5423 (to address issue #2700) made a change which installs a `SIGTERM` handler within `Task.__call__()`, which effectively ignores `SIGTERM`. This is...er...wrong, and leads to the following problems:

* Tasks cannot be called inline within a thread (eg common configurations of `uwsgi`, or even Django's `runserver`)
* The warm/cold shutdown distinction is erased
* The `REMAP_SIGTERM` configuration value is ignored (used in eg Heroku deployments)
* Signal handling by non-worker calling processes is forcibly overridden
* Conflicts with Celery's own worker signal handling (see [worker.py](https://github.com/celery/celery/blob/master/celery/apps/worker.py) and [platforms.py](https://github.com/celery/celery/blob/master/celery/platforms.py))

The PR in question was reverted in #5577, then un-reverted in #5586.

We've had to add a custom task class and override `__call__()` for production use to avoid this issue. Please re-revert #5423; if #2700 is still an issue inherent to Celery (as opposed to simply warm/cold shutdown differences), then a different fix should be considered.

Edit: after some deeper looks at the code, it seems as though this only applies for inline tasks under the base `Task` class, as `__call__()` is optimized away (see [trace.py](https://github.com/celery/celery/blob/master/celery/app/trace.py#L292)) for async/dispatched tasks if it hasn't been overridden. AFAICT it _would_ still be a problem if using a custom task class which overrides `__call__()` and calls `super()`, however.

# Checklist

- [x] I have read the relevant section in the
  [contribution guide](http://docs.celeryproject.org/en/latest/contributing.html#other-bugs)
  on reporting bugs.
- [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information

- [x] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.
- [ ] I have included the contents of ``pip freeze`` in the issue.
- [ ] I have included all the versions of all the external dependencies required
  to reproduce this bug.

#### Related Issues

- #2700 
- PR #5423 

#### Possible Duplicates

- None

## Environment & Settings

**Celery version**:

<details>
<summary><b><code>celery report</code> Output:</b></summary>
<p>

```
software -> celery:4.4.0rc3 (cliffs) kombu:4.6.4 py:3.7.4
            billiard:3.6.1.0 redis:3.3.7
platform -> system:Linux arch:64bit
            kernel version:4.9.184-linuxkit imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:redis results:redis://mps-redis:6379/0
```

</p>
</details>

# Steps to Reproduce

## Required Dependencies

* **Minimal Python Version**: N/A or Unknown
* **Minimal Celery Version**: 4.4.0rc2
* **Minimal Kombu Version**: N/A or Unknown
* **Minimal Broker Version**: N/A or Unknown
* **Minimal Result Backend Version**: N/A or Unknown
* **Minimal OS and/or Kernel Version**: N/A or Unknown
* **Minimal Broker Client Version**: N/A or Unknown
* **Minimal Result Backend Client Version**: N/A or Unknown

## Minimally Reproducible Test Case

See https://github.com/celery/celery/blob/master/celery/app/task.py#L396

# Expected Behavior

Should be able to call a task inline in a thread.

# Actual Behavior

When calling task inline under Django `runserver` (which uses threads):
```
Traceback (most recent call last):
  File "/app/pipelines/serializers.py", line 184, in create
    remote_worker=validated_data['with_remote'],
  File "/usr/local/lib/python3.7/site-packages/celery/local.py", line 191, in __call__
    return self._get_current_object()(*a, **kw)
  File "/usr/local/lib/python3.7/site-packages/celery/app/task.py", line 396, in __call__
    signal.signal(signal.SIGTERM, handle_sigterm)
  File "/usr/local/lib/python3.7/signal.py", line 47, in signal
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
ValueError: signal only works in main thread
```

**Comment by auvipy** [comment]
can you come with a PR to fix this?

**Comment by rneilson** [comment]
Should just be able to revert #5586 to remove the problem code (again; this'd be a re-re-revert). As far as #2700 goes, I'm not even sure it's a valid issue anymore -- it was opened back in 2015, and I believe the `acks_late`/warm-shutdown conflict was resolved in #926.
